activesupport/lib/active_support/inflector/methods.rb

```diff
@@ -195,12 +197,7 @@ def dasherize(underscored_word)
     #
     # See also #deconstantize.
     def demodulize(path)
-      path = path.to_s
```

Comment:
@jeremy Thanks for your comments. I've rebased the branch.

Must this method be responsible for the string coercion? After these changes this will not work anymore:

`ActiveSupport::Inflector.demodulize(:'Foo::Bar')`

or

`ActiveSupport::Inflector.demodulize(BigDecimal.new('1.00'))`

But I can't think of a use case for these examples. IHMO, it makes more sense to leave that responsibility outside this method.

cc @rafaelfranca